src/main/java/at/hannibal2/skyhanni/data/mob/Mob.kt

```diff
@@ -166,42 +174,44 @@ class Mob(
         fullEntityList().associateWith { this }
 
     internal fun internalAddEntity(entity: EntityLivingBase) {
+        internalRemoveColor()
         if (baseEntity.entityId > entity.entityId) {
             extraEntitiesList.add(0, baseEntity)
-            internalRemoveColor()
             baseEntity = entity
-            internalHighlight()
         } else {
             extraEntitiesList.add(extraEntitiesList.lastIndex + 1, entity)
         }
+        internalHighlight()
         updateBoundingBox()
         MobData.entityToMob[entity] = this
     }
 
     internal fun internalAddEntity(entities: Collection<EntityLivingBase>) {
         val list = entities.drop(1).toMutableList().apply { add(baseEntity) }
-        extraEntitiesList.addAll(0, list)
         internalRemoveColor()
+        extraEntitiesList.addAll(0, list)
         baseEntity = entities.first()
         internalHighlight()
         updateBoundingBox()
         removeExtraEntitiesFromChecking()
         MobData.entityToMob.putAll(entities.associateWith { this })
     }
 
-    internal fun internalUpdateOfEntity(entity: EntityLivingBase) = when (entity.entityId) {
-        baseEntity.entityId -> {
-            internalRemoveColor()
-            baseEntity = entity
-            internalHighlight()
-        }
-
-        armorStand?.entityId ?: Int.MIN_VALUE -> armorStand = entity as EntityArmorStand
-        else -> {
-            extraEntitiesList.remove(entity)
-            extraEntitiesList.add(entity)
-            Unit // To make return type of this branch Unit
+    internal fun internalUpdateOfEntity(entity: EntityLivingBase) {
+        internalRemoveColor()
+        when (entity.entityId) {
+            baseEntity.entityId -> {
+                baseEntity = entity
+            }
+
+            armorStand?.entityId ?: Int.MIN_VALUE -> armorStand = entity as EntityArmorStand
+            else -> {
+                extraEntitiesList.remove(entity)
+                extraEntitiesList.add(entity)
+                Unit // To make return type of this branch Unit
```

Comment:
can be removed. having return value of `when` be `Any` is fine as well/ no difference to `Unit` here

Reply:
But Intelji cries, so I changed it. Since it should be Unit semanticly.